types/src/bytesrepr.rs

```diff
@@ -108,6 +108,8 @@ pub enum Error {
     Formatting,
     /// Not all input bytes were consumed in [`deserialize`].
     LeftOverBytes,
+    /// The deserialized object did not pass validation.
+    Validation,
```

Comment:
I'm against doing that. It's now pushing the responsibility of handling this new error variant in all call sides affecting all users of `FromBytes` - including smart contract developers. Up until now we haven't put validation into `FromBytes` either and I'm against that as well.

Reply:
Sure, let's discuss 👍

Reply:
The currently existing `Formatting` variant is now used instead of `Validation`.